pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/relocation/SegmentRelocator.java

```diff
@@ -175,10 +175,6 @@ private void rebalanceTable(String tableNameWithType) {
       relocate = true;
       LOGGER.info("Relocating COMPLETED segments for table: {}", tableNameWithType);
     }
-    if (_enableLocalTierMigration) {
```

Comment:
this flag controls the controller side logic - checking segment current vs. target tier and issue segment reload msg.

Reply:
on the server side, this TierBasedSegmentDirectoryLoader should be used, instead of the default loader. The former one sets the correct tier on the segment for the controller side logic to check.